test/functional/feature_evm_createaccesslist.py

```diff
@@ -0,0 +1,163 @@
+                "r",
+                encoding="utf8",
+            ).read()
+        )["object"]
+
+        self.reserved_bytecode = json.loads(
+            open(
+                get_solc_artifact_path("dfi_reserved", "deployed_bytecode.json"),
+                "r",
+                encoding="utf8",
+            ).read()
+        )["object"]
+
+        # Generate chain
+        self.node.generate(150)
+        self.nodes[0].utxostoaccount({self.address: "1000@DFI"})
+
+        # Create token before EVM
+        self.node.createtoken(
+            {
+                "symbol": "USDT",
+                "name": "USDT token",
+                "isDAT": True,
+                "collateralAddress": self.address,
+            }
+        )
+        self.node.generate(1)
+        self.node.minttokens("10@USDT")
+        self.node.generate(2)
+
+        self.key_pair = EvmKeyPair.from_node(self.node)
+        self.key_pair2 = EvmKeyPair.from_node(self.node)
+
+        # enable EVM, transferdomain, DVM to EVM transfers and EVM to DVM transfers
+        self.nodes[0].setgov(
+            {
+                "ATTRIBUTES": {
+                    "v0/params/feature/evm": "true",
+                    "v0/params/feature/transferdomain": "true",
+                    "v0/transferdomain/dvm-evm/enabled": "true",
+                    "v0/transferdomain/dvm-evm/dat-enabled": "true",
+                    "v0/transferdomain/evm-dvm/dat-enabled": "true",
+                }
+            }
+        )
+        self.nodes[0].generate(2)
+        self.usdt = self.nodes[0].w3.eth.contract(
+            address=self.contract_address_usdt, abi=self.abi
+        )
+
+        self.node.transferdomain(
+            [
+                {
+                    "src": {"address": self.address, "amount": "1@USDT", "domain": 2},
+                    "dst": {
+                        "address": self.key_pair.address,
+                        "amount": "1@USDT",
+                        "domain": 3,
+                    },
+                    "singlekeycheck": False,
+                }
+            ]
+        )
+        self.node.generate(1)
+
+        assert_equal(
+            self.usdt.functions.balanceOf(self.key_pair.address).call()
+            / math.pow(10, self.usdt.functions.decimals().call()),
+            Decimal(1),
+        )
+
+        transfer_data = self.usdt.encodeABI(
+            fn_name="transfer",
+            args=[self.key_pair2.address, Web3.to_wei("0.5", "ether")],
+        )
+
+        transaction = {
+            "from": self.key_pair.address,
+            "to": self.contract_address_usdt,
+            "data": transfer_data,
+        }
+        access_list = self.node.eth_createAccessList(transaction)
+        al = access_list["accessList"][0]
+
+        assert_equal(len(al["storageKeys"]), 3)
+        assert_equal(al["address"], "0xff00000000000000000000000000000000000001")
+
+
+if __name__ == "__main__":
+    AccessListTest().main()
```

Comment:
Can we write test pipelines similar to other functional tests where we abstract test setup, and each functional test into methods? Each method should also do a rollback on the chain state so that we could add more test coverage quickly without the need for each test to be dependent on each other